### PR TITLE
add codecov upload action in coverage.yaml

### LINKED ISSUE #1196: Code Coverage Reports Not Uploading to Codecov, Stuck on Outdated Snapshot
Currently, our code coverage reports are not being uploaded to Codecov for test analysis. As a result, Codecov is still showing a year-old snapshot of our repository. The task is to resolve this upload issue and ensure that the reports reflect the latest coverage from the current head commit.
<details>
<summary>screenshot</summary>
Screenshot from 2024-10-07 19-56-35
</details>

### PR DESCRIPTION
**Pull request by Shourya742**
closes #1196

### CODE REVIEW

**Comment by Shourya742** [comment]
> **Tarpaulin Test ** Expected — Waiting for status to be reported

Trying to configure act locally to test the flow once.

File: .github/workflows/auto-rebase.yaml
Comment (plebhash) [inline .github/workflows/auto-rebase.yaml:1]: why are we deleting all these files?

**Review by GitGab19 — APPROVED** [review]

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -1,18 +1,17 @@
 # Performs test coverage of project's libraries using cargo-tarpaulin and generates results using codecov.io.
 # The following flags are set inside `tarpaulin.toml`:
 # `features = "..."`: Includes the code with the listed features. The following features result in a
-#     tarpaulin error and are NOT included: derive, alloc, arbitrary-derive, attributes, and
```
Comment (jbesraa) [inline .github/workflows/coverage.yaml:4]: Please clerify what have changed in this paragraph.
Also, I dont think you modified the `timeout` here to 1200s, which was changed here https://github.com/stratum-mining/stratum/pull/1197/commits/105acda841d67178846402129a96cd088678b471#diff-8965ddb70c4bbfcd7467189aba2a7d41675f85a1f5ff23119bea37504b96f351R4

File: .github/workflows/coverage.yaml [hunk 2]
```diff
@@ -39,12 +38,19 @@ jobs:
       - name: Generate code coverage
         run: |
           ./scripts/tarpaulin.sh
+      
+      - name: List files after tarpaulin run
+        run: |
+          echo "Checking files in protocols directory"
```
Comment (jbesraa) [inline .github/workflows/coverage.yaml]: What is this checking? 
could we also do `ls -l protocols && ls -l roles && ls -l utils` ?

File: roles/tarpaulin.toml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 features = "with_buffer_pool async_std tokio with_tokio default"
 run-types = [ "Lib" ]
 timeout = "120s"
-fail-under = 20
+fail-under = 0
```
Comment (jbesraa) [inline roles/tarpaulin.toml:5]: what does `file-under` mean here?

File: .github/workflows/coverage.yaml (hunk 2 above)
Comment (Shourya742) [reply .github/workflows/coverage.yaml]: This is an outdated snapshot

Comment (Shourya742) [reply .github/workflows/coverage.yaml]: It was used to check if the xml files were generating or not...

File: roles/tarpaulin.toml (hunk 3 above)
Comment (Shourya742) [reply roles/tarpaulin.toml:5]: `fail-under` is like if the coverage is less than this percentage, then fail

File: .github/workflows/coverage.yaml (hunk 1 above)
Comment (Shourya742) [reply .github/workflows/coverage.yaml:4]: Just formatting changed... that timeout is now restored to 120s

File: .github/workflows/coverage.yaml (hunk 2 above)
Comment (jbesraa) [reply .github/workflows/coverage.yaml]: Umm you got my confused (: 

Introducing changes in a commit and then reverting them in the next is not recommended. can you please fix up the second commit?

**Review by rrybarczyk — APPROVED** [review]

**Review by GitGab19 — APPROVED** [review]

**Pull request merged**